### ISSUE TITLE
Force "namespaced" extension names

### ISSUE THREAD

**Issue by SPQRInc** [pagekit/pagekit#884]
## Problem

As described [here](https://pagekit-forum.org/forum/thread/52-using-your-own-namespace-for-your-module-to-avoid-name-clashes/?postID=273&highlight=namespace#post273) it's very useful to create some kind of "namespaces" for extensions and themes. I think this should be forced by Pagekit, as some extension names like `blog` or `gallery` can cause name clashes. At least it should be forced in Pagekit `1.1.x`; until this version a notice should be shown.

## Technical Details

- Pagekit version: 1.0.13
- Webserver: Apache 2.4
- Database: SQLite
- PHP Version: 7.1

**Comment by MalteScharenberg** [comment]
Agree! The module name should always match the package name. We are planning to change this with the next minor update (as you suggested).

**Comment by tobbexiv** [comment]
First step for this: Update the examples / tutorials in the documentation as they all use non-namespaced extensions.

But I see a few issues with this:
- Namespacing the extension will not help for everything. You can still have the same tables / routes / shortcuts ... and checking everything might be too much. But updating the guidelines to ask devs to use namespaces might help. And every dev who is using proper namespaces should not have issues with any other extension or theme.
- How do you want to ensure that there is no installation of an extension with a non-namespaced name using upload functionality? With the approach as it is now you can only check the extension name after the package is installed. But it might be enough if we enforce in the marketplace that uploads are only possible with prefixed namespaces.

**Comment by SPQRInc** [comment]
That's correct - but it would be a first step to extend the docs. And if it's enforced on the marketplace, one could at least say: Software from marketplace satisfies some basic requirements. If you install software from third-party-websites, you should maybe doublecheck it.
